smart-contracts/contracts/cl-vault/src/vault/range.rs

```diff
@@ -333,6 +337,60 @@ pub fn do_swap_deposit_merge(
         },
     )?;
 
+    let calculate_swap: SwapCalculationResult = calculate_swap_amount(
+        deps,
+        env,
+        position_id,
+        balance0,
+        balance1,
+        target_lower_tick,
+        target_upper_tick,
+        twap_window_seconds,
+    )?;
+
+    // Start constructing the response
+    let mut response = Response::new()
+        .add_attribute("method", "reply")
+        // TODO: This is not really exact, this function is invoked by 2 different reply entrypoints so the action should be conditionally set
```

Comment:
While the function may be invoked in two different code paths, the action we do here is still pretty much the same, so I don't think this matters